src/types/lib/index.ts

```diff
@@ -3,10 +3,10 @@ import { weierstrass } from '../../utils/ec';
 import type { BigNumberish } from '../../utils/num';
 import { CompiledContract, CompiledSierraCasm, ContractClass } from './contract';
 
-// Common Signature Type which needs to be imported from weierstrass
-// and imported at many places
-// This is because stark.ts doesn't export SignatureType
-export type Signature = weierstrass.SignatureType;
+export type WeierstrassSignatureType = weierstrass.SignatureType;
+export type ArraySignatureType = string[];
+export type AnySignatureType = string | number | ArraySignatureType | WeierstrassSignatureType;
```

Comment:
I don't like this type tbh. This is too flexible. `ArraySignatureType | WeierstrassSignatureType` should be good for all cases.

Reply:
Agree, didn't like it either but wanted to make it smarter so users don't need to wrap it in an array. But as you suggested I will just drop it and force array for custom.